app/views/edit.erb

```diff
@@ -1,10 +1,10 @@
-<form action="/robots/<%= @robot.id %>" method="post">
+<form action="/robots/<%= robot_world.all.find{|robo|robo["name"]==@robot.name}["id"] %>" method="post">
```

Comment:
in general, you should avoid querying your database in your views. here, you're calling all and find to find a specific robot's id. this should live in the controller and be passed in using an instance variable 👍